src/main/verify.cpp

```diff
@@ -0,0 +1,210 @@
+doTryFetchTransaction(
+    boost::asio::steady_timer& timer,
+    Backend::CassandraBackend& backend,
+    ripple::uint256 const& hash,
+    boost::asio::yield_context& yield,
+    std::uint32_t const attempts = 0)
+{
+    try
+    {
+        return backend.fetchTransaction(hash, yield);
+    }
+    catch (Backend::DatabaseTimeout const& e)
+    {
+        if (attempts >= MAX_RETRIES)
+            throw e;
+
+        wait(timer, "Transaction read error");
+        return doTryFetchTransaction(timer, backend, hash, yield, attempts + 1);
+    }
+}
+
+static Backend::LedgerPage
+doTryFetchLedgerPage(
+    boost::asio::steady_timer& timer,
+    Backend::CassandraBackend& backend,
+    std::optional<ripple::uint256> const& cursor,
+    std::uint32_t const sequence,
+    boost::asio::yield_context& yield,
+    std::uint32_t const attempts = 0)
+{
+    try
+    {
+        return backend.fetchLedgerPage(cursor, sequence, 2000, false, yield);
+    }
+    catch (Backend::DatabaseTimeout const& e)
+    {
+        if (attempts >= MAX_RETRIES)
+            throw e;
+
+        wait(timer, "Page read error");
+        return doTryFetchLedgerPage(
+            timer, backend, cursor, sequence, yield, attempts + 1);
+    }
+}
+
+static const CassResult*
+doTryGetTxPageResult(
+    CassStatement* const query,
+    boost::asio::steady_timer& timer,
+    Backend::CassandraBackend& backend,
+    std::uint32_t const attempts = 0)
+{
+    CassFuture* fut = cass_session_execute(backend.cautionGetSession(), query);
+    CassResult const* result = cass_future_get_result(fut);
+    cass_future_free(fut);
+
+    if (result != nullptr)
+        return result;
+
+    if (attempts >= MAX_RETRIES)
+        throw std::runtime_error("Already retried too many times");
+
+    wait(timer, "Unexpected empty result from nft paging");
+    return doTryGetTxPageResult(query, timer, backend, attempts + 1);
+}
+
+static void
+verifyNFTs(
+    std::vector<NFTsData>& nfts,
+    Backend::CassandraBackend& backend,
+    boost::asio::yield_context& yield)
+{
+    if (nfts.size() <= 0)
+        return;
+
+    for(auto const& nft: nfts){
+        std::optional<Backend::NFT> writtenNFT = backend.fetchNFT(nft.tokenID, nft.ledgerSequence, yield);
+
+        if(!writtenNFT.has_value())
+            throw std::runtime_error("NFT was not written!");
+
+        Blob writtenUriBlob = writtenNFT->uri;
+        std::string writtenUriStr = ripple::strHex(writtenUriBlob);
+
+        auto fetchOldUri = nft.uri;
+        std::string oldUriStr = nft.uri.has_value() ? ripple::strHex(nft.uri.value()) : "";
+
+        if(oldUriStr.compare(writtenUriStr) != 0){
+            BOOST_LOG_TRIVIAL(warning) <<"\nNFTokenID "<< to_string(nft.tokenID) << " failed to match URIs!\n";  
+            throw std::runtime_error("Failed to match!");
+        }
+        else{
+            BOOST_LOG_TRIVIAL(info) <<"\nNFTokenID "<< to_string(nft.tokenID) << " URI matched!\n";         
```

Comment:
instead of logging for each success, what about logging at the end the total number of NFTs that were verified? i think that would be cleaner